docs/examples/Complex Numbers.ipynb

```diff
@@ -0,0 +1,318 @@
+    "                vals=ComplexNumbers())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `ComplexNumbers` validator requires explicitly complex values. Floats and ints will *not* pass."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Sorry, but 1.0 is not complex\n",
+      "Sorry, but -1.0 is not complex\n",
+      "Sorry, but 8.2 is not complex\n",
+      "Sorry, but 3.141592653589793 is not complex\n"
+     ]
+    }
+   ],
+   "source": [
+    "for value in np.array([1, -1, 8.2, np.pi]):\n",
+    "    try:\n",
+    "        imp(value)\n",
+    "        print(f'Succesfully set the parameter to {value}')\n",
+    "    except TypeError:\n",
+    "        print(f'Sorry, but {value} is not complex')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The easiest way to make a scalar complex is probably by adding `0j` to it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Succesfully set the parameter to (1+0j)\n",
+      "Succesfully set the parameter to (-1+0j)\n",
+      "Succesfully set the parameter to (8.2+0j)\n",
+      "Succesfully set the parameter to (3.141592653589793+0j)\n"
+     ]
+    }
+   ],
+   "source": [
+    "for value in np.array([1, -1, 8.2, np.pi]) + 0j:\n",
+    "    try:\n",
+    "        imp(value)\n",
+    "        print(f'Succesfully set the parameter to {value}')\n",
+    "    except TypeError:\n",
+    "        print(f'Sorry, but {value} is not complex')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Array-valued parameters\n",
+    "\n",
+    "There is no separate complex-valued array validator, since the `Arrays` validator can be customized to cover any real or complex valued case.\n",
+    "\n",
+    "Let's make a little array to hold some quantum state amplitudes. Let's pretend to be in a 5-dimensional Hilbert space. Our state parameter should thus hold 5 complex numbers (the state expansion coefficients in some implicit basis)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qcodes.utils.validators import Arrays"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The proper validator should accept complex numbers and should reject anything of the wrong shape. Note that we get to decide whether we want to accept \"non-strictly complex\" data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "amps_val_strict = Arrays(shape=(5,), valid_types=(np.complex128, np.complex64))\n",
```

Comment:
Should we use `np.complex` in the same way that we use the none size specific int and float types

Reply:
Certainly.